ShoppingList/ShoppingList/ItemVisibilityConverter.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Windows.Data;
+using System.Windows.Navigation;
+
+namespace ShoppingList
+{
+    public class ItemVisibilityConverter : IValueConverter
+    {
+        public object Convert(object value, Type? targetType,  object? parameter, CultureInfo? culture)
+        {
+            return value is Item ? "Visible" : "Hidden";

```

Comment:
You will want to return a Visibility recognized by WPF, as well as use "collapsed" instead of "hidden" because hidden will simply make a blank space and not effectively "hide" your editing elements. I know this seems counterintuitive based on the wording of "hidden", had me stumped for awhile. 

```suggestion
            return value is Item ? Visibility.Visible : Visibility.Collapsed;
```

Reply:
Agreed